File: backend/app/schemas/warehouse.py
from app.schemas.base import ORMModel
from pydantic import BaseModel


class WarehouseCreate(BaseModel):
    name: str
    code: str


class WarehouseUpdate(BaseModel):
    name: str | None = None
    code: str | None = None
    is_active: bool | None = None


class WarehouseRead(ORMModel):
    id: int
    name: str
    code: str
    is_active: bool

File: frontend/src/types/warehouse.ts
export type Warehouse = {
  id: number;
  name: string;
  code: string;
  is_active: boolean;
};

export type WarehouseCreate = {
  name: string;
  code: string;
};

export type WarehouseUpdate = Partial<WarehouseCreate> & {
  is_active?: boolean;
};
